x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichPolicy.java

```diff
@@ -202,7 +208,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeStringCollection(indices);
         out.writeString(matchField);
         out.writeStringCollection(enrichFields);
-        Version.writeVersion(elasticsearchVersion, out);
+        if (out.getTransportVersion().before(TransportVersions.ENRICH_ELASTICSEARCH_VERSION_REMOVED)) {
+            // emit the current version of elasticsearch for bwc serialization reasons
+            Version.writeVersion(Version.CURRENT, out);
```

Comment:
Wasn't a goal to remove the use of the `Version` class from `EnrichPolicy`? Should we just call out.writeVInt() here?

Reply:
I had exactly that written as WIP, but I thought these two remaining BWC/serialization oriented references to the `Version` class were more palatable than being too knowledgable of the details of the serialization of the `Version` class itself (and duplicating the implementation as trickery).

That said, though, it's just a slight preference of my own that this feels like the cleanest way to leave things -- I'm absolutely happy to take that next step if folks prefer it.

Reply:
We'll likely want to further clean this up, so as not to have loose Version.CURRENT still in the code (otherwise we don't know if that is a "real" use of Version that could contaminate serverless). However, I think we can do some of that as a followup.